Source/Core/DolphinQt/Debugger/BranchWatchDialog.cpp

```diff
@@ -135,15 +135,15 @@ bool BranchWatchProxyModel::filterAcceptsRow(int source_row, const QModelIndex&)
   if (!m_origin_symbol_name.isEmpty())
   {
     if (const QVariant& symbol_name_v = sourceModel()->GetSymbolList()[source_row].origin_name;
-        !symbol_name_v.isValid() ||
-        !symbol_name_v.value<QString>().contains(m_origin_symbol_name, Qt::CaseInsensitive))
+        !symbol_name_v.isValid() || !reinterpret_cast<const QString*>(symbol_name_v.data())
```

Comment:
No... this just feels wrong.

There must be a better solution than just randomly casting types.

Reply:
The "better solution" would be the [non-member `get` function](https://doc.qt.io/qt-6/qvariant.html#get) which was added in Qt6.6, but we are still using Qt6.5. Even if the `get` function was available to me, the only difference is [it does some runtime assertions that the variant is valid and handles edge cases](https://github.com/qt/qtbase/blob/5b151ea2d23dc3834180d3ec6495ac5d99cae550/src/corelib/kernel/qvariant.h#L631-L662) before doing a static_cast on the `data()` pointer (I mistakenly used reinterpret_cast here, which I will fix). I don't need these checks, since the the QVariant will always either contain a QString or be invalid.